node/src/neighborhood/mod.rs

```diff
@@ -1190,20 +1210,23 @@ impl Neighborhood {
         hostname_opt: Option<&str>,
     ) -> Option<Vec<&'a PublicKey>> {
         let mut minimum_undesirability = i64::MAX;
-        self.routing_engine(
-            vec![source],
-            0,
-            target_opt,
-            minimum_hops,
-            payload_size,
-            direction,
-            &mut minimum_undesirability,
-            hostname_opt,
-        )
-        .into_iter()
-        .filter(|cr| cr.undesirability <= minimum_undesirability)
-        .map(|cr| cr.nodes)
-        .next()
+        let result = self
+            .routing_engine(
+                vec![source],
+                self.compute_initial_undesirability(source, payload_size, direction),
+                target_opt,
+                minimum_hops,
+                payload_size,
+                direction,
+                &mut minimum_undesirability,
+                hostname_opt,
+            )
+            .into_iter()
+            .filter(|cr| cr.undesirability <= minimum_undesirability)
+            .map(|cr| cr.nodes)
```

Comment:
These two lines could be replaced by a single method and so done in a single operation:

a) filter_map(||),
b) flat_map(||)

With same guts:

```
|cr| match cr.undesirability <= minimum_undesirability {
        true => Some(cr.nodes),
        false => None
}
```